### ISSUE TITLE
nodeset boot doesn't update xnba .uefi/.elilo config files

### ISSUE THREAD

**Issue by kcgthb** [xcat2/xcat-core#4824]
Hi!

I'm not entirely sure if that's a bug or a configuration issue on my end, but it looks like `nodeset <node> boot` doesn't properly update the `/tftpboot/xcat/xnba/nodes/<node>.{elilo,uefi}` config files for stateful deployments.

For instance, when deploying a `netboot=xnba` node (via `rinstall`), I got:
```
# rinstall sh-101-60
Provision node(s): sh-101-60
# lsdef sh-101-60 -i netboot
Object name: sh-101-60
    netboot=xnba

# cat /tftpboot/xcat/xnba/nodes/sh-101-60
#!gpxe
#install centos7.4-x86_64-cpu
imgfetch -n kernel http://${next-server}/tftpboot/xcat/osimage/sherlock-compute_cpu/vmlinuz
imgload kernel
imgargs kernel quiet inst.repo=http://sh-hn02:80/install/centos7.4/x86_64 inst.ks=http://sh-hn02:80/install/autoinst/sh-101-60 ip=dhcp  inst.cmdline  console=tty0 console=ttyS0,115200   BOOTIF=01-${netX/machyp}
imgfetch http://${next-server}/tftpboot/xcat/osimage/sherlock-compute_cpu/initrd.img
imgexec kernel

# cat /tftpboot/xcat/xnba/nodes/sh-101-60.elilo
default="xCAT"
delay=0

image=/tftpboot/xcat/osimage/sherlock-compute_cpu/vmlinuz
   label="xCAT"
   initrd=/tftpboot/xcat/osimage/sherlock-compute_cpu/initrd.img
   append="quiet inst.repo=http://sh-hn02:80/install/centos7.4/x86_64 inst.ks=http://sh-hn02:80/install/autoinst/sh-101-60 ip=dhcp  inst.cmdline  console=tty0 console=ttyS0,115200   BOOTIF=%B"

# cat /tftpboot/xcat/xnba/nodes/sh-101-60.uefi
#!gpxe
chain http://${next-server}/tftpboot/xcat/elilo-x64.efi -C /tftpboot/xcat/xnba/nodes/sh-101-60.elilo
```

If I want to boot from the local disk, I do `nodeset sh-101-60 boot`, and the `<node>` config file is updated, but not `<node>.uefi` nor `<node>.elilo`:

```
# nodeset sh-101-60 boot
sh-101-60: boot

# cat /tftpboot/xcat/xnba/nodes/sh-101-60
#!gpxe
#boot
exit

# cat /tftpboot/xcat/xnba/nodes/sh-101-60.elilo
default="xCAT"
delay=0

image=/tftpboot/xcat/osimage/sherlock-compute_cpu/vmlinuz
   label="xCAT"
   initrd=/tftpboot/xcat/osimage/sherlock-compute_cpu/initrd.img
   append="quiet inst.repo=http://sh-hn02:80/install/centos7.4/x86_64 inst.ks=http://sh-hn02:80/install/autoinst/sh-101-60 ip=dhcp  inst.cmdline  console=tty0 console=ttyS0,115200   BOOTIF=%B"

# # cat /tftpboot/xcat/xnba/nodes/sh-101-60.uefi
#!gpxe
chain http://${next-server}/tftpboot/xcat/elilo-x64.efi -C /tftpboot/xcat/xnba/nodes/sh-101-60.elilo
```

Is that expected?

**Comment by immarvin** [comment]
hi @kcgthb , `nodeset boot ` will update the dhcp lease:
```
host bybc0607 {
  dynamic;
  deleted;
}
host bybc0607 {
  dynamic;
  hardware ethernet 42:14:0a:05:6a:07;
  uid 42:14:0a:05:6a:07;
  fixed-address 10.5.106.7;
        supersede server.ddns-hostname = "bybc0607";
        supersede host-name = "bybc0607";
        if option user-class-identifier = "xNBA" and option client-architecture
             = 00:00 {
          supersede server.filename =
                  "http://${next-server}/tftpboot/xcat/xnba/nodes/bybc0607";
        } elsif option client-architecture = 00:00 {
          supersede server.filename = "xcat/xnba.kpxe";
        } else {
          supersede server.filename = "";
        }
}
```

**Comment by kcgthb** [comment]
Aah, I see, thanks @immarvin 
So I take those files not being modified is not a problem.

Thanks!
